src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,132 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<T>(null, value, null);
+            tail = head;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.setNext(newNode);
+            tail = newNode;
+        }
+        size = size + 1;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> oldNode = getNode(index);
+            Node<T> newNode = new Node<>(oldNode.getPrev(), value, oldNode);
+            if (index != 0) {
+                oldNode.getPrev().setNext(newNode);
+            } else {
+                head = newNode;
+            }
+            oldNode.setPrev(newNode);
+            size = size + 1;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+        return getNode(index).getValue();
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index);
+        T returnValue = node.getValue();
+        node.setValue(value);
+        return returnValue;
```

Comment:
We need add runtime exception IndexOutOfBoundsException, if our index will be more than size or less than thero.
 public T set(T value, int index) {
     if ( index >= 0 && index < size) {
        Node<T> node = getNode(index);
        T returnValue = node.getValue();
        node.setValue(value);
        return returnValue;
        } else {
         throw new IndexOutOfBoundsException(" ");
         }

Reply:
This check is performed inside getNode(index) method which throws the exception